Token.py

```diff
@@ -21,12 +22,160 @@ def __init__(
         self.token_structure = token_structure
         self.category = category
 
+        # feature related fields
+        self.date_values = self.get_dates()
+        self.currency = self.get_currency()
+        self.address = self.get_address()
+        self.num_lable = self.get_num_label()
+        self.total_label = self.get_total_label()
+        self.date_label = self.get_date_label()
+        self.contains_digit = self.get_contains_digits()
+
     def __repr__(self):
         return self.text if self.text else str(self.token_structure)
 
     def __str__(self):
         return self.text if self.text else str(self.token_structure)
 
+    # returns true if has at least one digit, false otherwise
+    def get_contains_digits(self):
+        if self.text:
+            for c in self.text:
+                if c.isdigit():
+                    return True
+            return False
+
+    def get_date_label(self):
+        kw = ["date", "Date"]
+        if self.text:
+            for w in kw:
+                if w in self.text:
+                    return self.text
+
+    # tries to extract address from token
+    def get_address(self):
+        kw = [
```

Comment:
Suggested kw you could add:
- lorong, lor
- link
- valley
- view
- st (for street)
- west, south, north, east
- bukit
- tanjong

Reply:
ah yes thanks

Reply:
link might give false positives, same with st which i took out